core/vtk/ttkAlgorithm/ttkAlgorithm.cpp

```diff
@@ -403,6 +403,7 @@ void ttkAlgorithm::MPIPreconditioning(vtkDataSet *input) {
       static_cast<long int *>(ttkUtils::GetVoidPointer(
         input->GetPointData()->GetArray("GlobalPointIds"))));
   }
+  triangulation->preconditionDistributedVertices();
```

Comment:
thanks to that one liner, I believe *all* calls to `preconditionDistributedVertices()` can now be safely removed.